src/node/rpc/member_frontend.h

```diff
@@ -1152,6 +1152,137 @@ namespace ccf
       return get_proposal_info(proposal_id, proposal);
     }
 
+#ifdef ENABLE_JS_GOV
+
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wc99-extensions"
+
+    ccf::jsgov::ProposalSubmitted resolve_proposal(
+      kv::Tx& tx,
+      const ProposalId& proposal_id,
+      const std::string& proposal,
+      const std::string& constitution)
+    {
+      auto pi =
+        tx.rw<ccf::jsgov::ProposalInfoMap>("public:ccf.gov.proposals_info.js");
+      auto pi_ = pi->get(proposal_id);
+
+      std::vector<std::pair<MemberId, bool>> votes;
+      for (const auto& [mid, mb] : pi_->ballots)
+      {
+        std::string mbs = fmt::format(
+          "{}\n export default (proposal, proposer_id) => vote(proposal, "
+          "proposer_id);",
+          mb);
+
+        js::Runtime rt;
+        js::Context context(rt);
+        rt.add_ccf_classdefs();
+        js::populate_global_ccf(&tx, std::nullopt, nullptr, context);
+        auto ballot_func = context.function(
+          mbs, fmt::format("ballot from {} for {}", mid, proposal_id));
+
+        JSValue argv[2];
+        auto prop = JS_NewStringLen(context, proposal.c_str(), proposal.size());
+        argv[0] = prop;
+        auto pid = JS_NewStringLen(
+          context, pi_->proposer_id.data(), pi_->proposer_id.size());
+        argv[1] = pid;
+
+        auto val =
+          context(JS_Call(context, ballot_func, JS_UNDEFINED, 2, argv));
+        if (!JS_IsException(val))
+        {
+          votes.emplace_back(mid, JS_ToBool(context, val));
+        }
+        JS_FreeValue(context, ballot_func);
+        JS_FreeValue(context, prop);
+        JS_FreeValue(context, pid);
+      }
+
+      {
+        std::string mbs = fmt::format(
+          "{}\n export default (proposal, proposer_id, votes) => "
+          "resolve(proposal, proposer_id, votes);",
+          constitution);
+
+        js::Runtime rt;
+        js::Context context(rt);
+        js::populate_global_console(context);
+        rt.add_ccf_classdefs();
+        js::populate_global_ccf(&tx, std::nullopt, nullptr, context);
+        auto resolve_func =
+          context.function(mbs, fmt::format("resolve {}", proposal_id));
+        JSValue argv[3];
+        auto prop = JS_NewStringLen(context, proposal.c_str(), proposal.size());
+        argv[0] = prop;
+
+        auto prop_id = JS_NewStringLen(
+          context, pi_->proposer_id.data(), pi_->proposer_id.size());
+        argv[1] = prop_id;
+
+        auto vs = JS_NewArray(context);
+        size_t index = 0;
+        for (auto& [mid, vote] : votes)
+        {
+          auto v = JS_NewObject(context);
+          auto member_id = JS_NewStringLen(context, mid.data(), mid.size());
+          JS_DefinePropertyValueStr(
+            context, v, "member_id", member_id, JS_PROP_C_W_E);
```

Comment:
```suggestion
            context, v, "memberId", member_id, JS_PROP_C_W_E);
```

Reply:
I'm hesitant about this, because it's member_data, and is_operator and absolutely everything is_still_like_this. I think we do it all or we don't do it, but I don't want half-half.

Reply:
OK, not a big deal.